Added new command migrate:middleware-to-request-handler

michalbundyra opened the pull request:
Convert PSR-15 middleware to request handler in given path. It should be used to convert final middlewres binded to routes which are not using delegator/request handler.

Works only with PSR-15 middlewares (if you have interop middlewares migrate them first to PSR-15 middlewares by migrate:interop-middleware and then convert them to request handlers).

michalbundyra committed:
Added new command migrate:middleware-to-request-handler

Convert PSR-15 middleware to request handler in given path.
It should be used to convert final middlewres binded to routes
which are not using delegator/request handler.
Works only with PSR-15 middlewares (if you have interop middlewares
migrate them first to PSR-15 middlewares by migrate:interop-middleware
and then convert them to request handlers).

weierophinney commented:
I would argue that this should raise an exception if it detects any `$handler->handle` calls.

michalbundyra commented:
@weierophinney 
> I would argue that this should raise an exception if it detects any $handler->handle calls.

I'll try to add this check. I need to find what is the variable name and then look for `$var->handle` as you said.

michalbundyra committed:
Check if hanlder is used in migrate middleware to request handler tool

When we detect `$handler->handle(` phrase (where variable `$handler` can
be different and it's get from `process` method signature) we are not
going to do any changes in the file.

weierophinney commented on test/MigrateMiddlewareToRequestHandler/TestAsset/expected/MyMiddlewareWithHandler.php:
```diff
@@ -0,0 +1,21 @@
+<?php
+
```
We need a license docblock here and on the other test asset; we should likely also add the `declare(strict_types=1)`, though that's not completely necessary.

michalbundyra committed:
Added license docblock and strict type declaration in TestAssets

- per @weierophinney

weierophinney commented:
Thanks, @webimpress!
